src/store/getters/state-getters.ts

```diff
@@ -602,14 +607,21 @@ export const isSummaryMode = (state: StateIF): boolean => {
 
 /** Whether business name has changed. */
 export const hasBusinessNameChanged = (state: StateIF): boolean => {
-  const originalLegalName = getEntitySnapshot(state)?.businessInfo?.legalName
-  return (state.stateModel.nameRequest?.legalName !== originalLegalName)
+  const currentLegalName = state.stateModel.nameRequest.legalName ||
+    getBusinessInformation(state).legalName
+  const originalLegalName = getEntitySnapshot(state)?.businessInfo?.legalName ||
+    getCorrectedFiling(state)?.business?.legalName
```

Comment:
I added some fallbacks for BEN IA corrections. Eventually these will store a business snapshot (instead of saving the corrected filing) and these fallbacks can be removed.

I think it will be easy to find the fallbacks to remove: basically anything that looks at the corrected filing (or original IA).